### PR TITLE
net: openthread: Added config for OpenThread version selection.

### PR DESCRIPTION
**Pull request by kkasperczyk-no**
Config options used for selecting OpenThread stack version
were added.

Signed-off-by: Kamil Kasperczyk <kamil.kasperczyk@nordicsemi.no>

### CODE REVIEW

**Comment by kkasperczyk-no** [comment]
@jukkar could you take a look on it, as @rlubos is out of the office for some time?

**Review by jukkar — APPROVED** [review]
LGTM

**Review by carlescufi — CHANGES_REQUESTED** [review]
Please change `west.yml` to point to https://github.com/zephyrproject-rtos/openthread/pull/23

**Review by carlescufi — APPROVED** [review]

File: west.yml [hunk 1]
```diff
@@ -107,7 +107,7 @@ manifest:
       revision: 170a2579dd890f78f5056f0959cdb9c9bea259a1
       path: modules/lib/loramac-node
     - name: openthread
-      revision: d216ec32cdb78ec873689234f710b3c9bb096bcd
+      revision: pull/23/head
```
Comment (carlescufi) [inline west.yml:110]: Replace with e294e1d5073f40d7d735b40bda5bfbe05dbf7a8a

**Pull request merged**